pkg/segment/query/processor/queryprocessor.go

```diff
@@ -112,6 +112,11 @@ func newQueryProcessorHelper(queryType structs.QueryType, input streamer,
 	}, nil
 }
 
+func (dp *DataProcessor) IsGenerateDataProcessor() bool {
+	_, ok := dp.processor.(*gentimesProcessor)
+	return ok
+}
```

Comment:
This isn't the right file to add this method. It should be in dataprocessor.go by the other DataProcessor methods if you need this function

Reply:
addressed in the latest commit.